Move addr broadcasting from codegen layer to native layer.

gchanan opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #37615 Port existing zero_dim_dispatch optimizations from codegen and remove codegen capability.
* #37614 Kill broadcasting from the codegen layer.
* #37613 Move addmm broadcasting code from codegen layer to native layer.
* #37603 Move addbmm broadcasting from the codegen layer to native layer.
* #37582 Specify _th_ ops in CUDAUnaryOps macros so they are easier to find.
* #37547 Kill the ability to codegen tensor-based broadcasting.
* **#37546 Move addr broadcasting from codegen layer to native layer.**
* #37545 Move broadcasting code for fmod, fmod_ from codegen layer.
* #37544 Move baddbmm broadcasting from codegen layer to native layer.

Differential Revision: [D21315040](https://our.internmc.facebook.com/intern/diff/D21315040)

gchanan committed:
Move addr broadcasting from codegen layer to native layer.

[ghstack-poisoned]

bhosmer commented on aten/src/ATen/native/LinearAlgebra.cpp:
```diff
@@ -141,7 +141,9 @@ static void check_1d(const Tensor& t, const char* arg, const char* fn) {
   check_1d(vec2, "vec2", "addr");
-  return at::_addr(self, vec1, vec2, beta, alpha);
+  Tensor b_self;
+  std::tie(b_self) = expand_size(self, {vec1.size(0), vec2.size(0)}, "addr");
```
I see that `expand_size(self, sizes)` returns a 1-tuple but I'm not sure whyy

gchanan replied:
oh it's just for codegen simplicity, so the codegen can just always be written in terms of tuples.  We could change that later, not sure if anything else is calling into that function.

gchanan committed:
Update on "Move addr broadcasting from codegen layer to native layer."


Differential Revision: [D21315040](https://our.internmc.facebook.com/intern/diff/D21315040)

[ghstack-poisoned]

gchanan committed:
Update on "Move addr broadcasting from codegen layer to native layer."


Differential Revision: [D21315040](https://our.internmc.facebook.com/intern/diff/D21315040)

[ghstack-poisoned]

gchanan committed:
Update on "Move addr broadcasting from codegen layer to native layer."


Differential Revision: [D21315040](https://our.internmc.facebook.com/intern/diff/D21315040)

[ghstack-poisoned]
